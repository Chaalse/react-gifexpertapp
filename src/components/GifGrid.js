import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({categoria}) => {
    //Código obsoleto. Primera aproximación haciendo el componente
    // const [images, setImages] = useState([]);
    // useEffect(() => {
        
        //     getGifs( categoria )
        //     .then( imgs => setImages( imgs))
        
        // }, [ categoria])
        
        const {data:images, loading} = useFetchGifs( categoria);
    

    return (
        <>
        <h3>{ categoria}</h3>

        {   loading && <p>Loading</p>}

        <div className="card-grid">
                {
                    images.map(img =>(
                        <GifGridItem 
                        key = {img.id}
                        {...img}
                        />
                ))
                }
        </div>
        </>
    )
}
