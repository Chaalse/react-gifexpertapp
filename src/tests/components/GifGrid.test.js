import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');
describe('Pruebas de la funcionalidad GifGrid', () => {
    
    const categoria = 'Samurai Jack';
    
    test('Debe mostrar el componente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        
        const wrapper  =shallow(<GifGrid categoria = {categoria} />);
        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('Debe mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {

        const gifs =  [
            {
            id: 'ABC', 
            url: 'https://localhost/cualquier/cosa.jpg', 
            title: 'Cualquier Imagen',
        },
            {
            id: 'DEF', 
            url: 'https://localhost/cualquier/cosa.jpg', 
            title: 'Cualquier Imagen',
        },
        
        ];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper  =shallow(<GifGrid categoria = {categoria} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);       
        expect(wrapper.find('GifGridItem').length).toBe( gifs.length ); 
    });
    
    
});
