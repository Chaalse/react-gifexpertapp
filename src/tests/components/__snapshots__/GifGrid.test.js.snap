// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de la funcionalidad GifGrid Debe mostrar el componente 1`] = `
<Fragment>
  <h3>
    Samurai Jack
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas de la funcionalidad GifGrid Debe mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3>
    Samurai Jack
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Imagen"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Cualquier Imagen"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
